File: Java/Technology Fundamentals with Java/Exercises Java Code/src/MethodExercise/FactorialDivision.java
package MethodExercise;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class FactorialDivision {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        int firstNum = Integer.parseInt(reader.readLine());
        int secondNum = Integer.parseInt(reader.readLine());

        long firstNumFactorial = factorial(firstNum);
        long secondNumFactorial = factorial(secondNum);

        double result = 1.00 * firstNumFactorial / secondNumFactorial;

        System.out.printf("%.2f", result);
    }

    private static long factorial(int number) {
        long result = 1;

        for (int i = 2; i <= number; i++) {
            result *= i;
        }
        return result;
    }
}

File: C#/Technology Fundamentals with C#/04. Methods - Functions - Exercise/08. Factorial Division/FactorialDivision.cs
using System;

public class FactorialDivision
{
    public static void Main()
    {
        int firstNumber = int.Parse(Console.ReadLine());
        int secondNumber = int.Parse(Console.ReadLine());

        long factorialFirstNumber = factorial(firstNumber);
        long factorialSecondNumber = factorial(secondNumber);
        double result = 1.0 * factorialFirstNumber / factorialSecondNumber;
        Console.WriteLine($"{result:F2}");
    }

    private static long factorial(int firstNumber)
    {
        long factorial = 1;
        for (int i = 2; i <= firstNumber; i++)
        {
            factorial *= i;
        }
        return factorial;
    }
}
